tests/libvmi/lifecycle.go

```diff
@@ -0,0 +1,46 @@
+package libvmi
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+)
+
+func SetupVMI(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
```

Comment:
i think that `By` are really helpful to know where you stand, as check points, the rest is some logs and expecter dump which make it less comfortable to know where you are in case of a failure

Reply:
You should know based on where the assertion has occurred in the test.
Debugging using printf/s in tests is not something I am looking for for sure. I am familiar with several languages and test frameworks, none used printf/s in such a way.
We can open this for a discussion on a different media, such a discussion here will overload this PR  conversation.